src/Resources/Resources/help/Test-AzDeployment.md

```diff
@@ -12,104 +12,188 @@ Validates a deployment.
 ```
 
-### ByTemplateUriWithNoParameters
+### ResourceGroupWithTemplateObjectAndParameterFile
 ```
-Test-AzDeployment -Location <String> -TemplateUri <String> [-SkipTemplateParameterPrompt]
- [-ApiVersion <String>] [-Pre] [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+Test-AzDeployment -ScopeType <DeploymentScopeType> -ResourceGroupName <String> [-Mode <DeploymentMode>]
+ [-RollbackToLastDeployment] [-RollBackDeploymentName <String>] -TemplateParameterFile <String>
+ -TemplateObject <Hashtable> [-SkipTemplateParameterPrompt] [-ApiVersion <String>] [-Pre]
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+### ResourceGroupWithTemplateFileAndParameterObject
+```
+Test-AzDeployment -ScopeType <DeploymentScopeType> -ResourceGroupName <String> [-Mode <DeploymentMode>]
+ [-RollbackToLastDeployment] [-RollBackDeploymentName <String>] -TemplateParameterObject <Hashtable>
+ -TemplateFile <String> [-SkipTemplateParameterPrompt] [-ApiVersion <String>] [-Pre]
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+### ResourceGroupWithTemplateFileAndParameterFile
+```
+Test-AzDeployment -ScopeType <DeploymentScopeType> -ResourceGroupName <String> [-Mode <DeploymentMode>]
+ [-RollbackToLastDeployment] [-RollBackDeploymentName <String>] -TemplateParameterFile <String>
+ -TemplateFile <String> [-SkipTemplateParameterPrompt] [-ApiVersion <String>] [-Pre]
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+### ResourceGroupWithTemplateObjectAndNoParameters
+```
+Test-AzDeployment -ScopeType <DeploymentScopeType> -ResourceGroupName <String> [-Mode <DeploymentMode>]
+ [-RollbackToLastDeployment] [-RollBackDeploymentName <String>] -TemplateObject <Hashtable>
+ [-SkipTemplateParameterPrompt] [-ApiVersion <String>] [-Pre] [-DefaultProfile <IAzureContextContainer>]
+ [<CommonParameters>]
+```
+
+### ResourceGroupWithTemplateFileWithAndNoParameters
+```
+Test-AzDeployment -ScopeType <DeploymentScopeType> -ResourceGroupName <String> [-Mode <DeploymentMode>]
+ [-RollbackToLastDeployment] [-RollBackDeploymentName <String>] -TemplateFile <String>
+ [-SkipTemplateParameterPrompt] [-ApiVersion <String>] [-Pre] [-DefaultProfile <IAzureContextContainer>]
+ [<CommonParameters>]
 ```
 
 ## DESCRIPTION
 The **Test-AzDeployment** cmdlet determines whether a deployment template and its parameter values are valid.
 
 ## EXAMPLES
 
-### Example 1: Test deployment with a custom template and parameter file
+### Example 1: Test deployment at subscription scope with a custom template and parameter file
 ```
-PS C:\> Test-AzDeployment -Location "West US" -TemplateFile "D:\Azure\Templates\EngineeringSite.json" -TemplateParameterFile "D:\Azure\Templates\EngSiteParms.json"
+PS C:\> Test-AzDeployment -ScopeType "Subscription" -Location "West US" -TemplateFile "D:\Azure\Templates\EngineeringSite.json" -TemplateParameterFile "D:\Azure\Templates\EngSiteParms.json"
```

Comment:
@Tiano2017 same comment about adding the corresponding output to all examples in this file